examples/cosmos/chain_miscellaneous_test.go

```diff
@@ -440,6 +443,40 @@ func testTXFailsOnBlockInclusion(ctx context.Context, t *testing.T, chain *cosmo
 	require.Error(t, err)
 }
 
+func testTXEncodeDecode(ctx context.Context, t *testing.T, chain *cosmos.CosmosChain, users []ibc.Wallet) {
+	generate := chain.GetNode().TxCommand(users[0].KeyName(), "bank", "send", users[0].FormattedAddress(), users[1].FormattedAddress(), "1"+chain.Config().Denom, "--generate-only")
+
+	// check if generate contains "--gas"
+	require.Contains(t, generate, "--gas")
+
+	txJson, _, err := chain.GetNode().Exec(ctx, generate, nil)
+	require.NoError(t, err)
+
+	// encode
+	err = chain.GetNode().WriteFile(ctx, txJson, "tx.json")
+	require.NoError(t, err)
+	encode := chain.GetNode().TxCommand(users[0].KeyName(), "encode", chain.GetNode().HomeDir()+"/tx.json")
```

Comment:
should use path.Join vs +, resolved with push

Reply:
Good catch!